Add wxKeyEvent.IsAutoRepeat()

mjessome opened the pull request:
This adds the ability to detect whether a KEY_DOWN event is due to OS-level auto repeat, or is a user-initiated KEY_DOWN event.

This patch is coming from desire for this functionality on a project I am working on rather than manually tracking UP/DOWN events, and it has also been asked a number of times on the developer forums.

The flag is readily available on OSX/Cocoa, Qt and MSW, though in the research I've done, I can't seem to find any way to do this with GTK.

I have added a "Repeat" column to samples/keyboard to demonstrate its use.

Thanks!

vadz commented on interface/wx/event.h:
```diff
@@ -1547,6 +1547,14 @@ class wxKeyEvent : public wxEvent,
+    /**
+        Returns true if this event is an auto-repeat of the key, false if this
+        is the initial key press.
+
```
```suggestion

@since 3.1.6
```

vadz commented on src/osx/cocoa/window.mm:
```diff
@@ -450,8 +450,7 @@ long wxOSXTranslateCocoaKey( NSEvent* event, int eventType )
     wxevent.m_rawCode = [nsEvent keyCode];
     wxevent.m_rawFlags = modifiers;
-
-    wxevent.SetTimestamp( (int)([nsEvent timestamp] * 1000) ) ;
```
Just to confirm, was removing the call to `SetTimestamp()` intentional?

mjessome replied:
Excellent catch, my apologies -- This was entirely accidental.

vadz reviewed (dismissed):
Thanks a lot, the PR looks great and I'll merge it once the CI builds pass, just adding the single `@since` line.

mjessome committed:
Add wxKeyEvent.IsAutoRepeat()

Supported in Qt, Cocoa, MSW ports.
Keyboard example updated with a "Repeat" column.

mjessome commented:
Addressed both comments, thanks for taking a look!

vadz commented on src/msw/window.cpp:
```diff
@@ -6303,6 +6303,8 @@ MSWInitAnyKeyEvent(wxKeyEvent& event,
     event.m_rawFlags = (wxUint32) lParam;
     event.SetTimestamp(::GetMessageTime());
+
+    event.m_isRepeat = ((HIWORD(lParam) & KF_REPEAT) == KF_REPEAT;
```
I didn't notice it, but luckily the CI builds did: there is an extra parenthesis here:
```suggestion
    event.m_isRepeat = (HIWORD(lParam) & KF_REPEAT) == KF_REPEAT;
```

vadz reviewed (changes requested):
Need to fix the build.

vadz commented:
OK, I've fixed wxMSW myself and will merge it soon.

mjessome commented:
Thank you @vadz
